janusgraph-core/src/main/java/org/janusgraph/core/JanusGraphFactory.java

```diff
@@ -374,12 +384,11 @@ private static ReadConfiguration getLocalConfiguration(File file) {
                                   Pattern.quote(INDEX_CONF_FILE.getName()) +  ")"
             + ")");
 
-            final Iterator<String> keysToMangle = StreamSupport.stream(
+            final List<String> keysToMangle = StreamSupport.stream(
                 Spliterators.spliteratorUnknownSize(configuration.getKeys(), 0), false)
-                .filter(key -> null != key && p.matcher(key).matches()).iterator();
+                .filter(key -> null != key && p.matcher(key).matches()).collect(Collectors.toList());
```

Comment:
This will lead to java.util.ConcurrentModificationException. In commons-configuration2, it is not allowed to modify the configuration while iterating the keys at the same time.

Reply:
Thanks for the explanation! Make sense